.github/workflows/integration_tests.yml

```diff
@@ -0,0 +1,53 @@
+name: End to End integration tests
+
+on:
+  - pull_request
+  - push
+  - workflow_dispatch
+
+jobs:
+  integration-tests:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout rucio containers repository
+        uses: actions/checkout@v2
+        with:
+          repository: rucio/containers
+      - uses: actions/checkout@v2
+        name: Checkout rucio source
+        with:
+          path: dev/rucio
+      - name: Prepare Images
+        shell: bash
+        run: |
+          sed -i "s/RUN git clone .*/COPY .\/rucio \/tmp\/rucio/" $GITHUB_WORKSPACE/dev/Dockerfile_py3
+      - name: Build Images
+        uses: whoan/docker-build-with-cache-action@v5
+        id: build
+        with:
+          username: "${{ github.actor }}"
+          password: "${{ secrets.GITHUB_TOKEN }}"
+          registry: docker.pkg.github.com
+          context: "${{ github.workspace }}/dev"
+          dockerfile: Dockerfile_py3
+          image_name: rucio-dev
+      - name: Prepare Docker Compose
+        shell: bash
+        run: |
+          IMG=$(echo "${{ steps.build.outputs.FULL_IMAGE_NAME }}" | sed 's_/_\\/_g')
+          sed "s/image: rucio\/rucio-dev/image: $IMG/" $GITHUB_WORKSPACE/dev/rucio/etc/docker/dev/docker-compose-storage.yml
+      - name: Start containers
+        run: docker-compose -f $GITHUB_WORKSPACE/dev/rucio/etc/docker/dev/docker-compose-storage.yml up -d
+      - name: Initialize tests
+        shell: bash
+        run: |
+          docker exec -t dev_rucio_1 cp etc/rse-accounts.cfg.template etc/rse-accounts.cfg
+          docker exec -t dev_rucio_1 tools/run_tests_docker.sh -ir
+      - name: File Upload/Download Test
+        run: docker exec -t dev_rucio_1 pytest -v --full-trace lib/rucio/tests/test_rucio_server.py
```

Comment:
there was a slight change in how the tests are run now: `python -bb -m pytest -vvvrxs …`. I will definitely replace this cmdline with a documented helper tool soon, so that can be reused instead of this. So this change is optional on your side.

Reply:
done :)